web/src/lib/utils/asset-utils.ts

```diff
@@ -438,6 +440,26 @@ export const deleteStack = async (stackIds: string[]) => {
   }
 };
 
+export const keepThisDeleteOthers = async (keepAsset: AssetResponseDto, stack: StackResponseDto) => {
+  const $t = get(t);
+
+  try {
+    const assetsToDeleteIds = stack.assets.filter((asset) => asset.id !== keepAsset.id).map((asset) => asset.id);
+    await deleteStacks({ bulkIdsDto: { ids: [stack.id] } });
+    await deleteAssets({ assetBulkDeleteDto: { ids: assetsToDeleteIds } });
```

Comment:
It's better to switch the order of these two requests. If the delete fails, it will avoid the stack becoming unstacked.

Reply:
Sorry for the delay. Good suggestion. Updated in next commit.